File: src/Kaizen.Ui.Angular/src/app/comments/comments.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { CommentCreateContract, CommentItem, CommentFilter, CommentUpdateContract, CommentDeleteContract } from '../shared/models/comments.model';
import { Page } from '../shared/models/shared.models';
import { HttpParamsHelper } from '../shared/services/utils.service';

@Injectable()
export class CommentsService {

	constructor(private httpClient: HttpClient) {
	}

	public create(contract: CommentCreateContract): Observable<CommentItem> {
		return this.httpClient.post<CommentItem>('/api/comment', contract);
	}

	public query(filter: CommentFilter): Observable<Page<CommentItem>> {
		return this.httpClient.get<Page<CommentItem>>('/api/comment/list', { params: HttpParamsHelper.toHttpParams(filter) });
	}

	public update(contract: CommentUpdateContract): Observable<CommentItem> {
		return this.httpClient.put<CommentItem>('/api/comment', contract);
	}

	public delete(contract: CommentDeleteContract): Observable<any> {
		return this.httpClient.delete('/api/comment', { params: HttpParamsHelper.toHttpParams(contract) });
	}
}

File: src/Kaizen.Comments/Kaizen.Comments.Api/Services/CommentsService.cs
using JetBrains.Annotations;
using Kaizen.ApiGateway.Controllers;
using Kaizen.Comments.Api.Create;
using Kaizen.Comments.Api.Delete;
using Kaizen.Comments.Api.Filter;
using Kaizen.Comments.Api.Items;
using Kaizen.Comments.Api.Update;
using Kaizen.Common.Service;
using MassTransit;
using System.Threading.Tasks;

namespace Kaizen.Comments.Api.Services
{
	public class CommentsService : BusBasedService, ICommentsService
	{
		public CommentsService(IBusControl busControl) : base(busControl)
		{
		}

		public Task<CommentItem> Create([NotNull] CommentCreateContract contract)
		{
			return DoRequest<CommentCreateContract, CommentItem>(contract);
		}

		public Task<EmptyResponse> Delete([NotNull] CommentDeleteContract contract)
		{
			return DoRequest<CommentDeleteContract, EmptyResponse>(contract);
		}

		public Task<Page<CommentItem>> Query([NotNull] CommentFilter filter)
		{
			return DoRequest<CommentFilter, Page<CommentItem>>(filter);
		}

		public Task<CommentItem> Update([NotNull] CommentUpdateContract contract)
		{
			return DoRequest<CommentUpdateContract, CommentItem>(contract);
		}
	}
}
